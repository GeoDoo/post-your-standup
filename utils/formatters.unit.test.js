const { TEXT } = require('../constants')
const { today, formatIssues } = require('./formatters')

jest.mock('moment', () => () => ({
  format: () => '31-01-2020',
  fromNow: () => '6 months ago',
}))

describe('Helpers', () => {
  describe('today', () => {
    it("should provide the today's date in a nice format", () => {
      expect(today()).toBe('31-01-2020')
    })
  })

  describe('formatIssues', () => {
    it('should provide standup issues in a nice format', () => {
      const data = [
        {
          key: 'key',
          fields: {
            summary: 'summary',
            status: { name: 'status' },
            updated: 10021230120,
          },
        },
      ]
      const projectUrl = 'https://test.atlassian.net'

      expect(formatIssues(data, projectUrl)).toMatchSnapshot()
    })

    it('should provide useful feedback when no issues found', () => {
      const data = []

      expect(formatIssues(data)).toBe(TEXT.COMMANDS.STANDUP.NO_ISSUES_FOUND)
    })
  })
})
