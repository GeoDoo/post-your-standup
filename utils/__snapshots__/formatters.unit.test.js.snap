// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers formatIssues should provide standup issues in a nice format 1`] = `
"1.  <https://test.atlassian.net/browse/key|summary>

      Status: *undefined*

      _Last updated 6 months ago_
"
`;
